lib/vdsm/storage/hsm.py

```diff
@@ -2857,6 +2859,33 @@ def measure(self, sdUUID, imgUUID, volUUID, dest_format, backing=True,
 
         return dict(result=result)
 
+    @public
+    def prune_bitmaps(self, sdUUID, imgUUID, volUUID, baseUUID):
+        """
+        Prune stale bitmaps from the base volume.
+        Warning: Internal use only (i.e., from virt code).
+
+        Arguments:
+            sdUUID (str): The UUID of the storage domain that owns the volume.
+            imgUUID (str): The UUID of the image contained on the volume.
+            volUUID (str): The UUID of the top volume.
+            baseUUID (str): The UUID of the volume that will get its bitmaps
+                pruned if missing or invalid in top.
+        """
+        img_ns = rm.getNamespace(sc.IMAGE_NAMESPACE, sdUUID)
+        locks = [rm.Lock(sc.STORAGE, sdUUID, rm.SHARED),
+                 rm.Lock(img_ns, imgUUID, rm.EXCLUSIVE)]
+        with guarded.context(locks):
+            dom = sdCache.produce_manifest(sdUUID=sdUUID)
+            try:
+                base_vol = dom.produceVolume(imgUUID=imgUUID, volUUID=baseUUID)
+                top_vol = dom.produceVolume(imgUUID=imgUUID, volUUID=volUUID)
```

Comment:
In produce_manifest() and produceVolume() there no need to use key=value in
such call, both arguments are required, and the key=value style adds no value
because the name of the argument is self documentating.